1-exercises/B-highlight-words/script.js

```diff
@@ -1,5 +1,30 @@
 function highlightWords(paragraph, colours) {
   // Write your code here...
+  let displayEl = document.getElementById("content");
+  let pEl = document.createElement("p");
+  displayEl.appendChild(pEl);
+  let selectEl = document.createElement("select");
+  displayEl.appendChild(selectEl);
+
+  colours.forEach((colour) => {
+    let optEl = document.createElement("option");
+    optEl.textContent = colour;
+    optEl.value = colour;
+    selectEl.appendChild(optEl);
+  });
+
+  let wordsArr = paragraph.split(" ");
+  wordsArr.forEach((word) => {
+    let spanEl = document.createElement("span");
+    spanEl.innerText = `${word} `;
+    spanEl.addEventListener("click", highlight);
+    pEl.appendChild(spanEl);
+  });
+
+function highlight() {
```

Comment:
So I have two problems with this listener, firstly it isn't quite finished, which is a pity, it's  the last piece of the puzzle, secondly it's nested inside the highlightWords function, this is not ideal.

There's a good example or two on this page, but some of the text is a bit dense  https://developer.mozilla.org/en-US/docs/Web/Events/Event_handlers